syntax = "proto3";
package pb;

import "asset.proto";

message MapCellTile {
	Tileset.Name name = 1;
	uint32 id = 2;
}

message MapCell {
	uint32 id = 1;
	MapCellTile tile = 2;
}

message MapTrigger {
	uint32 id = 1;
	oneof trigger {
		MapTriggerGate gate = 2;
		MapTriggerGate exit = 3;
		MapTriggerWater water = 4;
	}
}

message MapTriggerGate {
	Zone.Name zone = 1;
	float x = 2;
	float y = 3;
}

message MapTriggerWater {
	float drag = 1;
}

message MapMob {
	string def = 1;
	float x = 2;
	float y = 3;
	float scale = 4;
}

message Map {
	enum Name {
		unknown = 0;
		map2 = 2;
		map3 = 3;
	}
	Map.Name name = 1;
	uint32 w = 2;
	uint32 h = 3;
	repeated MapCell terrain = 4;
}

message Zone {
	enum Name {
		unknown = 0;
		zone2 = 2;
		zone3 = 3;
	}
	Zone.Name name = 1;
	Map.Name map = 2;
	repeated MapTrigger trigger = 3;
	repeated MapMob mob = 4;
}
